docs/examples/ratify-verify-azure-cmd.md

```diff
@@ -1,296 +1,361 @@
+
+az aks create \
+    --resource-group "${GROUP_NAME}" \
+    --name "${AKS_NAME}" \
+    --node-vm-size Standard_DS3_v2 \
+    --node-count 1 \
+    --generate-ssh-keys \
+    --enable-workload-identity \
+    --attach-acr ${ACR_NAME} \
+    --enable-oidc-issuer
+
+# Connect to the AKS cluster:
+az aks get-credentials --resource-group ${GROUP_NAME} --name ${AKS_NAME}
+
+export AKS_OIDC_ISSUER="$(az aks show -n ${AKS_NAME} -g ${GROUP_NAME} --query "oidcIssuerProfile.issuerUrl" -otsv)"
+```
+
+6. This step above may take around 10 minutes to complete. The registration status can be checked by running the following command:
+
+```bash
+az feature show --namespace "Microsoft.ContainerService" --name "EnableWorkloadIdentityPreview" -o table
+Name                                                      RegistrationState
+--------------------------------------------------------    -------------------
+Microsoft.ContainerService/EnableWorkloadIdentityPreview    Registered
+```
+
+7. Establish federated identity credential. On AZ CLI `${RATIFY_NAMESPACE}` is where you deploy Ratify:
+
+```bash
+az identity federated-credential create \
+  --name ratify-federated-credential \
+  --identity-name "${IDENTITY_NAME}" \
+  --resource-group "${GROUP_NAME}" \
+  --issuer "${AKS_OIDC_ISSUER}" \
+  --subject system:serviceaccount:"${RATIFY_NAMESPACE}":"gatekeeper-system"
+```
+
+8. Create an Azure Key Vault and set an access policy. If you have an AKV, you can skip this step.
+
+```bash
+az keyvault create \
+  --resource-group ${GROUP_NAME} \
+  --name ${KEYVAULT_NAME}
+```
+
+9. Set the environmental variable for Azure Key Vault URI.
+
+```bash
+export VAULT_URI=$(az keyvault show --name ${KEYVAULT_NAME} --resource-group ${GROUP_NAME} --query "properties.vaultUri" -otsv)
+```
+
+10. Import your own private key and certificates. You can import it on the portal as well.
+
+```bash
+az keyvault certificate import \
+  --vault-name ${KEYVAULT_NAME} \
+  -n <Certificate Name> \
+  -f /path/to/certificate
+```
+ 
+11. Configure policy for user-assigned managed identity:
+    
+```bash
+az keyvault set-policy --name ${KEYVAULT_NAME} \
+  --certificate-permissions get \
+  --object-id ${IDENTITY_OBJECT_ID}
+```
+
+## Deploy Gatekeeper and Ratify on AKS 
+
+1. Deploy Gatekeeper from helm chart:
+
+```bash
+helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts
+
+helm install gatekeeper/gatekeeper  \
+  --name-template=gatekeeper \
+  --namespace gatekeeper-system --create-namespace \
+  --set enableExternalData=true \
+  --set validatingWebhookTimeoutSeconds=5 \
+  --set mutatingWebhookTimeoutSeconds=2
+```
+
+2. Install Ratify on AKS from helm chart:
+
+```bash
+# Add a Helm repo
+helm repo add ratify https://deislabs.github.io/ratify
+# Download the notary verification certificate
+curl -sSLO https://raw.githubusercontent.com/deislabs/ratify/main/test/testdata/notary.crt
+# Install Ratify
+helm install ratify \
```

Comment:
@FeynmanZhou I believe the purpose of the guide is to verify against the newly created image and signature created in the Notation signing portion of the guide. This install step uses the quick start published test image `wabbitnetworks.azurecr.io` and its test signature rather than the new user defined one. We need to update this install step with the correct AKV, Workload Identity flags for install.

Reply:
Here's the install step from our Azure CI test: https://github.com/deislabs/ratify/blob/8ed9ccf4075afaf8936a44fcc67b2c769043aae3/scripts/azure-ci-test.sh#L66

Reply:
Good catch. I mistakenly copied the helm installation step from this PR. So should it be:

```
```bash
git clone https://github.com/deislabs/ratify.git

curl -sSLO https://raw.githubusercontent.com/deislabs/ratify/main/test/testdata/notary.crt

helm install ratify ./charts/ratify --atomic \
  --namespace ${RATIFY_NAMESPACE} \
  --set oras.authProviders.azureWorkloadIdentityEnabled=true \
  --set azureWorkloadIdentity.clientId=${IDENTITY_CLIENT_ID} \
  --set akvCertConfig.enabled=true \
  --set akvCertConfig.vaultURI=${VAULT_URI} \
  --set akvCertConfig.cert1Name=${NOTARY_PEM_NAME} \
  --set akvCertConfig.tenantId=${TENANT_ID}
```

Reply:
no need to git clone or curl the notary.crt part either. The install command looks good. I would add the `--create-namespace` flag too just in case:

```
helm install ratify \
    ./charts/ratify --atomic \
    --namespace ${RATIFY_NAMESPACE} --create-namespace \
    --set akvCertConfig.enabled=true \
    --set akvCertConfig.vaultURI=${VAULT_URI} \
    --set akvCertConfig.cert1Name=${NOTARY_PEM_NAME} \
    --set akvCertConfig.tenantId=${TENANT_ID} \
    --set oras.authProviders.azureWorkloadIdentityEnabled=true \
    --set azureWorkloadIdentity.clientId=${IDENTITY_CLIENT_ID}
```

Reply:
Accepted and updated